### PR TITLE
Fix duplicate `groups` in kwargs of Conv2d

### PR DESCRIPTION
**Pull request by jeffreysijuntan**
## Types of changes

- [x ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs change / refactoring / dependency upgrade

## Motivation and Context / Related issue

This pull request addresses the following two issues:
1. In current `__patched_conv2d_ops` in `CUDALongTensor`, we assume the default `group` is 1. If the user pass in `group` as kwargs, then we will encounter duplicate kwargs error. This diff addresses this issue by explicitly checking whether `group` is passed in as kwargs.
2. Specify the `device` of kernel for average pooling in `gradients.py`. If `device` is not specified, then running `Avgpool2d` layer on GPU will throw a `device doesn't match` error.

## How Has This Been Tested (if it applies)
Tested training of LeNet on GPU. 

## Checklist

- [ ] The documentation is up-to-date with the changes I made.
- [ ] I have read the **CONTRIBUTING** document and completed the CLA (see **CONTRIBUTING**).
- [ ] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

**Review by knottb — APPROVED** [review]

**Comment by knottb** [comment]
Appears that CircleCI is showing failures in benchmarks due to the MNIST server being down.

This is not a problem with the code here but is blocking our ability to merge this PR for now. The status of the MNIST server can be tracked [here](https://github.com/pytorch/vision/issues/3554).

I'm rerunning the CircleCI test to see if we can get this landed.

**Comment by jeffreysijuntan** [comment]
> Appears that CircleCI is showing failures in benchmarks due to the MNIST server being down.
> 
> This is not a problem with the code here but is blocking our ability to merge this PR for now. The status of the MNIST server can be tracked [here](https://github.com/pytorch/vision/issues/3554).
> 
> I'm rerunning the CircleCI test to see if we can get this landed.

No problem. Thanks!

**Comment by knottb** [comment]
Looks like `conv_transpose2d` is failing now. Going to look into this a bit more to gather information.

```
~ Test test.test_arithmetic.TestArithmetic.test_conv2d_rectangular_image_many_channels failed ~
===========
Exceptions:
===========
** Process 0 ** :
Traceback (most recent call last):
  File "/home/circleci/CrypTen/test/multiprocess_test_case.py", line 199, in _run
    getattr(self, test_name)()
  File "/home/circleci/CrypTen/test/test_arithmetic.py", line 560, in test_conv2d_rectangular_image_many_channels
    self._conv2d((16, 7), 5)
  File "/home/circleci/CrypTen/test/test_arithmetic.py", line 622, in _conv2d
    self._check(encr_conv, reference, "%s failed" % func_name)
  File "/home/circleci/CrypTen/test/test_arithmetic.py", line 55, in _check
    self.assertTrue(test_passed, msg=msg)
  File "/usr/local/lib/python3.7/unittest/case.py", line 705, in assertTrue
    raise self.failureException(msg)
AssertionError: False is not true : conv_transpose2d failed

** Process 1 ** :
Traceback (most recent call last):
  File "/home/circleci/CrypTen/test/multiprocess_test_case.py", line 199, in _run
    getattr(self, test_name)()
  File "/home/circleci/CrypTen/test/test_arithmetic.py", line 560, in test_conv2d_rectangular_image_many_channels
    self._conv2d((16, 7), 5)
  File "/home/circleci/CrypTen/test/test_arithmetic.py", line 622, in _conv2d
    self._check(encr_conv, reference, "%s failed" % func_name)
  File "/home/circleci/CrypTen/test/test_arithmetic.py", line 55, in _check
    self.assertTrue(test_passed, msg=msg)
  File "/usr/local/lib/python3.7/unittest/case.py", line 705, in assertTrue
    raise self.failureException(msg)
AssertionError: False is not true : conv_transpose2d failed
```

File: crypten/cuda/cuda_tensor.py [hunk 1]
```diff
@@ -197,13 +197,20 @@ def __patched_conv_ops(op, x, y, *args, **kwargs):
...
+        else:
+            kwargs["groups"] = nb2
+
         z_encoded = getattr(torch, op)(
-            x_enc_span, y_enc_span, *args, **kwargs, groups=nb2
+            x_enc_span, y_enc_span, *args, **kwargs
         )
-        z_encoded = z_encoded.reshape(bs, nb2, c_z, *z_encoded.size()[2:]).transpose_(
+
+        groups = kwargs["groups"] // nb2 if op in ["conv_transpose1d", "conv_transpose2d"] else 1
```
Comment (knottb) [inline crypten/cuda/cuda_tensor.py:212]: `conv_transpose2d` groups is giving incorrect result in unit test:

`test.test_arithmetic.TestArithmetic.test_conv2d_rectangular_image_many_channel`

**Review by knottb — CHANGES_REQUESTED** [review]

**Pull request closed without merging**